tests/unittests/graphOptzTest.cpp

```diff
@@ -161,6 +161,33 @@ TEST_F(GraphOptz, optimizeBatchNormAfterConvFP16) {
   EXPECT_TRUE(llvm::isa<SaveNode>(save));
 }
 
+/// Check that transpose constant folding is done before BatchNorm optimization,
+/// which allows to merge BatchNorm into Convolution with transposed weights.
+TEST_F(GraphOptz, optimizeBatchNormAfterConvWithTransposedWeights) {
+  Node *input =
+      mod_.createPlaceholder(ElemKind::FloatTy, {1, 10, 20, 3}, "A", false);
+  auto *filter =
+      mod_.createPlaceholder(ElemKind::FloatTy, {16, 3, 5, 5}, "filter", true);
```

Comment:
not sure why these are trainable :) any reason?

Reply:
No particular reason, the optimization should work with both.
Is there any reason to use un-trainable placeholder vs trainable in tests? Both kinds seem to be used randomly in tests...

Reply:
it's just when I see that var is trainable I start to think whether there are any reasons behind that, or whether it has to be trainable.